magicctapipe/scripts/lst1_magic/lst1_magic_dl2_to_dl3.py

```diff
@@ -108,6 +136,27 @@ def load_dl2_data_file(input_file, quality_cuts, irf_type):
     n_events = len(df_events.groupby(['obs_id', 'event_id']).size())
     logger.info(f'--> {n_events} stereo events')
 
+    # Calculate the dead time correction factor:
+    logger.info('\nCalculating the dead time correction factor...')
+
+    condition = '(time_diff > 0) & (time_diff < 0.1)'
+
+    time_diffs_lst = df_events.query(f'(tel_id == 1) & {condition}')['time_diff'].to_numpy()
+    time_diffs_magic = df_events.query(f'(tel_id == 2) & {condition}')['time_diff'].to_numpy()
```

Comment:
M1 and M2 dead time is strongly correlated so it is fine to take only one as you do, but maybe we can make it more "democratic", such that it would work also in the case we have observations when M1 was not working at all.
How about counting the M1 and M2 events and using for this one the one with more events?